src/H5FDsubfiling/H5subfiling_common.c

```diff
@@ -188,23 +187,27 @@ H5_get_subfiling_object(int64_t object_id)
             assert((size_t)obj_index < sf_context_cache_size);
         }
 
-        /*
-         * Since this cache currently just keeps all entries until
-         * application exit, context entry indices should just be
-         * consecutive
-         */
-        assert((size_t)obj_index <= sf_context_cache_num_entries);
-        if ((size_t)obj_index < sf_context_cache_num_entries)
-            ret_value = sf_context_cache[obj_index];
-        else {
-            assert(!sf_context_cache[sf_context_cache_num_entries]);
+        ret_value = sf_context_cache[obj_index];
+        if (!ret_value) {
+            size_t next_idx;
 
             /* Allocate a new subfiling context object */
-            if (NULL == (ret_value = calloc(1, sizeof(subfiling_context_t))))
+            if (NULL == (sf_context_cache[obj_index] = calloc(1, sizeof(subfiling_context_t))))
```

Comment:
What's the benefit of swapping out ret_value here?

Reply:
No benefit, it's just a preference. I prefer to assign the return value of malloc/calloc to something first, then set ret_value rather than the other way around.